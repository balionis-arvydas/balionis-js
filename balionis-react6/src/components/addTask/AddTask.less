// @import "colors.less";
.addTaskWrapper {
    border: 1px solid #ff0000;
}
