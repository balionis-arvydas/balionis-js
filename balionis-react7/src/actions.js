import c from "./constants";
import helpers from "./helpers";

const open = async (testURL, options = {}) => {
    const page = await browser.newPage();
    const url = testURL || helpers.getGlobalHomeURL();
    await page.goto(url, options);
    return page;
};

const login = async (testURL, waitForRedirect) => {
    jest.setTimeout(c.LOGIN_TIMEOUT);
    const page = await open(testURL, { timeout: c.LOGIN_TIMEOUT });

    console.info("login: url=" + page.url());

    if (waitForRedirect) {
        await page.waitForNavigation();
    }

    await page.waitForSelector(c.LOGIN_SELECTOR, { timeout: c.LOGIN_TIMEOUT });

    return page;
};

export default {
    open,
    login,
}
