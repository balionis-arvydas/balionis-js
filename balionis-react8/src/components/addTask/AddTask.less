// @import "colors.less";
.addTaskWrapper {
    border: 1px solid #ff0000;
    display: flex;

    .addTaskInput {
        flex: 1 1 auto;
    }

    .addTaskButton {
        flex: 0 0 auto;
    }
}

