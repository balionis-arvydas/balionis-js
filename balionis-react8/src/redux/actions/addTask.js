import { ADD_TASK } from "./actionTypes";

let nextTaskId = 0;

export const addTask = (content, taskId) => ({
    type: ADD_TASK,
    payload: {
        id: (taskId ? taskId : ++nextTaskId),
        content,
    }
});
