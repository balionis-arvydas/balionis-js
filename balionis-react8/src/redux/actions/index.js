export * from './addTask';
export * from './removeTask';
