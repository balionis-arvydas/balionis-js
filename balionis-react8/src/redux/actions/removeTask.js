import { REMOVE_TASK } from "./actionTypes";

export const removeTask = id => ({
    type: REMOVE_TASK,
    payload: {
        id
    }
});
