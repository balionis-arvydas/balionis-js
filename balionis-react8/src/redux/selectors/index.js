// export { default as taskList } from './taskList';
export * from './taskList';
