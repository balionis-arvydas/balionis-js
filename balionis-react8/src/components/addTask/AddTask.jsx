import React from "react";
import { connect } from "react-redux";

import { mapStateToProps, mapDispatchToProps } from "./AddTaskProperties";

import "./AddTask.less";

class AddTask extends React.Component {
    constructor (props) {
        super(props);

        this.state = {
            value: "",
            disabled: true,
        };
    }

    handleChange (event) {
        const value = event.target.value;
        const disabled = value === null || value.length < 3;
        this.setState({
            value,
            disabled
        });
    }

    handleSubmit (event) {
        event.preventDefault();
        this.props.addTask(this.state.value);
        this.setState({
            value: '',
            disabled: true,
        });
    }

    render () {
        return (
            <div className="addTaskWrapper">
                <input type="text"
                    value={this.state.value}
                    onChange={(event) => this.handleChange(event)}
                    className="addTaskInput"/>
                <button onClick={(event) => this.handleSubmit(event)}
                    disabled={this.state.disabled}
                    className="addTaskButton">
                    Add
                </button>
            </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddTask);
