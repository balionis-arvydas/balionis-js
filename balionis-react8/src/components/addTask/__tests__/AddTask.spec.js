import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mockStore } from '../../../redux/__tests__/mockStore';

import AddTask from '../AddTask';
import { addTask } from "../../../redux/actions";

describe('AddTask', () => {
    let store;

    beforeAll(() => {
        jest.resetAllMocks();
    });

    beforeEach(() => {
        store = mockStore({
            taskList: { tasks: [] }
        });
        store.dispatch = jest.fn();
    });

    test("snapshot", async () => {
        const component = renderer.create(
            <Provider store={store}>
                <AddTask/>,
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test("should add task", async () => {
        const component = mount(
            <Provider store={store}>
                <AddTask/>
            </Provider>
        );

        component.find('input').simulate('change', { target: { value: '12345' } });
        component.find('button').simulate('click');

        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            addTask('12345', 1)
        );
    });
});
