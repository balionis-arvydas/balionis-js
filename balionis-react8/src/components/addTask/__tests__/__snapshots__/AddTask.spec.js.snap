// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask snapshot 1`] = `
Array [
  <div
    className="addTaskWrapper"
  >
    <input
      className="addTaskInput"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="addTaskButton"
      disabled={true}
      onClick={[Function]}
    >
      Add
    </button>
  </div>,
  ",",
]
`;
